packages/component_code_gen/config/config.py

```diff
@@ -1,21 +1,48 @@
+import os
 from dotenv import load_dotenv
 load_dotenv()
 
-import os
+
+DEFAULTS = {
+    "OPENAI_API_TYPE": "openai",
+    "OPENAI_MODEL": "gpt-4",
+    "OPENAI_API_VERSION": "2023-05-15",
+    "LOGGING_LEVEL": "WARN",
+    "ENABLE_DOCS": False
+}
+
+
+def get_env_var(var_name, required=False):
+    if os.environ.get(var_name):
+        return os.environ.get(var_name)
+    if required and var_name not in DEFAULTS:
```

Comment:
Totally fine to keep some defaults here but I would recommend keeping each of these only in one place:

* Defaults are fine where we don't need to set user-specific env, e.g. the model, openAI version, etc. should probably be set here and not in env.
* Request any user-specific vars be set in `.env. and yell at people (by raising an error) if they aren't set.